Nodejs/Product/Npm/SemverVersion.cs

```diff
@@ -31,13 +31,13 @@ public struct SemverVersion {
             "^(?<major>[0-9]+)"
             + "\\.(?<minor>[0-9]+)"
             + "\\.(?<patch>[…0-9]+)" // The '…' is there to handle the 'classy' library, which has a very long version number - see slightly snarky comment about that unadulterated bag of hilarity below.
-            + "(?:-(?<prerelease>[…0-9A-Za-z-]+(\\.[…0-9A-Za-z-]+)*))?"
-            + "(?:\\+(?<buildmetadata>[…0-9A-Za-z-]+(\\.[…0-9A-Za-z-]+)*))?$",
-            RegexOptions.IgnoreCase | RegexOptions.Singleline);
+            + "(?:-(?<prerelease>[…0-9A-Za-z-.]+))?"
+            + "(?:\\+(?<buildmetadata>[…0-9A-Za-z-.]+))?$",
+            RegexOptions.Singleline);
 
         private static readonly Regex RegexOptionalFragment = new Regex(
-            "^[…0-9A-Za-z-]+(\\.[…0-9A-Za-z-]+)*$",
-            RegexOptions.IgnoreCase | RegexOptions.Singleline);
+            "^[…0-9A-Za-z-.]+$",

```

Comment:
Same here and on line 35. The prior pattern of `^[…0-9A-Za-z-]+(\\.[…0-9A-Za-z-]+)*` looks right to me in terms of disallowing invalid formats. `^[…0-9A-Za-z-.]+` seems a little too lax. Why is this change needed?